### PR TITLE
new dapp tutorial with unity game engine 

### PR DESCRIPTION
**Pull request by lucasespinosa28**
## Description
I wrote  this tutorial how  to use [usedapp](https://usedapp.readthedocs.io/en/latest/) with unity for a [hackathon](https://gitcoin.co/hackathon/0x-hack/) on Gitcoin

## Related Issue
this is a new tutorial

### CODE REVIEW

**Review by lucasespinosa28 — COMMENTED** [review]
tutorial added

File: src/content/developers/tutorials/Using-UseDApp-with-unity/index.md [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+title: Using UseDApp with Unity
+description: Learn how to use UseDApp to interact with Unity engine project 
+author: "lucasespinosa28"
+tags: ["usedapp", "unity","react","dapp"]
+skill: intermediate
+lang: en
+sidebar: true
+sourceUrl: https://github.com/lucasespinosa28/usedappwithUnity
+---
+## Introduction {#introduction}
```
Comment (wackerow) [inline src/content/developers/tutorials/Using-UseDApp-with-unity/index.md:11]: can remove this first header (automatically gives an H1 with the `title` above)

File: src/content/developers/tutorials/Using-UseDApp-with-unity/index.md [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+In this tutorial, you will learn how to use [UseDApp](https://usedapp.readthedocs.io/en/latest/) a React plugins for rapid Dapp development, 
+and how to interact with a Unity using react, you will get your account balance inside a Unity and send a transaction from unity.
+
+### You need have [Unity](https://unity.com/) installed
```
Comment (wackerow) [inline src/content/developers/tutorials/Using-UseDApp-with-unity/index.md:15]: Would make this just a bold statement wrapped in double-asterisks instead of an empty header section

File: src/content/developers/tutorials/Using-UseDApp-with-unity/index.md [hunk 3]
```diff
@@ -0,0 +1,206 @@
...
+description: Learn how to use UseDApp to interact with Unity engine project 
+author: "lucasespinosa28"
+tags: ["usedapp", "unity","react","dapp"]
+skill: intermediate
+lang: en
+sidebar: true
+sourceUrl: https://github.com/lucasespinosa28/usedappwithUnity
+---
+## Introduction {#introduction}
+In this tutorial, you will learn how to use [UseDApp](https://usedapp.readthedocs.io/en/latest/) a React plugins for rapid Dapp development,
```
Comment (wackerow) [inline src/content/developers/tutorials/Using-UseDApp-with-unity/index.md:12]: Check capitalization consistancy... "dapp", "React", "Unity"

File: src/content/developers/tutorials/Using-UseDApp-with-unity/index.md [hunk 4]
```diff
@@ -0,0 +1,206 @@
...
+sourceUrl: https://github.com/lucasespinosa28/usedappwithUnity
+---
+## Introduction {#introduction}
+In this tutorial, you will learn how to use [UseDApp](https://usedapp.readthedocs.io/en/latest/) a React plugins for rapid Dapp development, 
+and how to interact with a Unity using react, you will get your account balance inside a Unity and send a transaction from unity.
+
+### You need have [Unity](https://unity.com/) installed 
+
+## Unity project {#unity-project}
+On the Unity will have a text to receive the account balance, and a button to send a transaction using UseDApp,
```
Comment (wackerow) [inline src/content/developers/tutorials/Using-UseDApp-with-unity/index.md:18]: _Inline_ code snippets should be wrapped in single backticks instead of triple (triple for code blocks).

Check the grammar on this as well... not fully clear what the instructions here are

File: src/content/developers/tutorials/Using-UseDApp-with-unity/index.md [hunk 5]
```diff
@@ -0,0 +1,206 @@
...
+to communicate from unity to [UseDApp](https://usedapp.readthedocs.io/en/latest/) we will need an extra code using ```jslib```.
+
+### 1\. Create a new unity project and configure to webgl {#new-unity-project}
+Create a new unity project, the name of the project and if the project is 2d or 3d makes no difference.
```
Comment (wackerow) [inline src/content/developers/tutorials/Using-UseDApp-with-unity/index.md:22]: Grammar check here please... and also, given it's a tutorial, we may want to just guide the user into 2D instead of leaving it up to them just for simplicity.  We can then remove the "makes no difference"

File: src/content/developers/tutorials/Using-UseDApp-with-unity/index.md [hunk 6]
```diff
@@ -0,0 +1,206 @@
...
+
+### You need have [Unity](https://unity.com/) installed 
+
+## Unity project {#unity-project}
+On the Unity will have a text to receive the account balance, and a button to send a transaction using UseDApp,
+to communicate from unity to [UseDApp](https://usedapp.readthedocs.io/en/latest/) we will need an extra code using ```jslib```.
+
+### 1\. Create a new unity project and configure to webgl {#new-unity-project}
+Create a new unity project, the name of the project and if the project is 2d or 3d makes no difference. 
+![Captura de Tela (80)](https://user-images.githubusercontent.com/52639395/119444261-4e3a6980-bd01-11eb-9946-e7b7653918b3.png)
```
Comment (wackerow) [inline src/content/developers/tutorials/Using-UseDApp-with-unity/index.md:23]: If we could download any external images and place into the same folder (after renaming to something coherent, all kebab/lowercase), then reference using `Alternate text explaining image`

File: src/content/developers/tutorials/Using-UseDApp-with-unity/index.md [hunk 7]
```diff
@@ -0,0 +1,206 @@
...
+
+Configure the project to build to WebGL to use in the browser, go to ```File > build settings``` choose Webgl, and click on the switch platform.
+
+![Captura de Tela (78)](https://user-images.githubusercontent.com/52639395/119444075-0a476480-bd01-11eb-8b00-a29c13a0b468.png)
+
+After switch to webgl go to ```Edit > Project Settings > Player ```, in publish setting choose disable compression format
+
+![Captura de Tela (77)](https://user-images.githubusercontent.com/52639395/119443979-e2f09780-bd00-11eb-9eea-2bed9ca5596d.png)
+
+### 2\. Create the interface{#create-UI}
```
Comment (wackerow) [inline src/content/developers/tutorials/Using-UseDApp-with-unity/index.md:33]: space before `{#create-UI}`, and can remove escape slashes (`### 2. `)

File: src/content/developers/tutorials/Using-UseDApp-with-unity/index.md (hunk 6 above)
Comment (wackerow) [reply src/content/developers/tutorials/Using-UseDApp-with-unity/index.md:23]: (Same with the others)

File: src/content/developers/tutorials/Using-UseDApp-with-unity/index.md [hunk 8]
```diff
@@ -0,0 +1,206 @@
...
+Now create the UI to interact with unity
+  1. Create a ```Canvas```.
+  2. Create a ```Text``` with name **BalanceText** to store the balance.
+  3. Create a ```Button```to call method to send a transaction and a ```Input Field``` to store the address for transaction.
+![Captura de Tela (90)](https://user-images.githubusercontent.com/52639395/119541875-4dd3ba00-bd65-11eb-913a-ba320f0dfc55.png)
+
+### 3\. Write code to receive the balance {#unity-balance}
+In asset folder create a new cs file ```BalanceController.cs```, drag and drop this file into gameobject ```BalanceText```,open the code file and writer the code below.
```
Comment (wackerow) [inline src/content/developers/tutorials/Using-UseDApp-with-unity/index.md:41]: space after second comma

File: src/content/developers/tutorials/Using-UseDApp-with-unity/index.md [hunk 9]
```diff
@@ -0,0 +1,206 @@
...
+    {
+        GameObject.Find("BalanceText").GetComponent<Text>().text = $"Balance: {amoun} EHT";
+    }
+```
+The **EtherBalanceToUnity** will receive the account balance from react app.
+
+### 4\. Write code to send a transaction {#unity-send-transaction}
+In asset folder create a new folder with name **Plugins**, create a new file ```script.jslib``` inside this new folder, 
+[jslib](https://docs.unity3d.com/Manual/webgl-interactingwithbrowserscripting.html) script are similar  to javascript,
+this script build a bridge between your Unity project and the react app,**SendTranscationFromUnitya** is the function that will exist in both unity and react app,write the code below.
```
Comment (wackerow) [inline src/content/developers/tutorials/Using-UseDApp-with-unity/index.md:53]: double-check spacing... extra space after "similar", missing space before "**SendTranscationFromUnitya**" which also has a stray "a" that needs removal at the end. Also missing space before last comma ("app, write").

Doubled check capitalization too... "JavaScript", "Unity", "React" are all names

File: src/content/developers/tutorials/Using-UseDApp-with-unity/index.md [hunk 10]
```diff
@@ -0,0 +1,206 @@
...
+```javascript
+    mergeInto(LibraryManager.library, {
+      SendTranscationFromUnity:function (address,amount) {
+        ReactUnityWebGL.SendTranscationFromUnity(Pointer_stringify(address), amount);
+      },
+});
+```
+Create a new file ```SendController.cs``` put the code in the gameobject **Button**,open the file and add this code.
```
Comment (wackerow) [inline src/content/developers/tutorials/Using-UseDApp-with-unity/index.md:61]: (space after comma)

File: src/content/developers/tutorials/Using-UseDApp-with-unity/index.md [hunk 11]
```diff
@@ -0,0 +1,206 @@
...
+        //A valid eth address is 42 characters long 
+        if (input.text.Length == 42)
+        {
+          //Send transaction information to react app.
+          SendTranscationFromUnity(input.text, 1e15);
+        }
+    }
+```
+
+Now our unit project is ready, wait for the next tutorial part to build the project, the code only works when is builded, it will not work in unity editor mode,
```
Comment (wackerow) [inline src/content/developers/tutorials/Using-UseDApp-with-unity/index.md:88]: Consider something like:
"Now our Unity project is ready. In the next step we will build the project to allow us to interact with the project."

File: src/content/developers/tutorials/Using-UseDApp-with-unity/index.md [hunk 12]
```diff
@@ -0,0 +1,206 @@
...
+        if (input.text.Length == 42)
+        {
+          //Send transaction information to react app.
+          SendTranscationFromUnity(input.text, 1e15);
+        }
+    }
+```
+
+Now our unit project is ready, wait for the next tutorial part to build the project, the code only works when is builded, it will not work in unity editor mode, 
+to learning more about unity in web browser and how interact using javascript consult the [official documentation](https://docs.unity3d.com/Manual/webgl-gettingstarted.html)
```
Comment (wackerow) [inline src/content/developers/tutorials/Using-UseDApp-with-unity/index.md:89]: `[More on getting started with WebGL.](https://docs.unity3d.com/Manual/webgl-gettingstarted.html)`

File: src/content/developers/tutorials/Using-UseDApp-with-unity/index.md [hunk 13]
```diff
@@ -0,0 +1,206 @@
...
+
+## React app with UseApp {#React-UseDApp}
+Create a react app that use UseDApp,with UseDapp is possible easy connect you web wallet in a dapp to send transaction and get information about your account,
```
Comment (wackerow) [inline src/content/developers/tutorials/Using-UseDApp-with-unity/index.md:92]: We should break this into multiple sentences as comma use is confusing and hard to follow

File: src/content/developers/tutorials/Using-UseDApp-with-unity/index.md [hunk 14]
```diff
@@ -0,0 +1,206 @@
...
+also use [react-unity-webgl](https://github.com/jeffreylanters/react-unity-webgl) to help use the Unity in a react app.
```
Comment (wackerow) [inline src/content/developers/tutorials/Using-UseDApp-with-unity/index.md:93]: "to help use **the Unity** in a react app." 
Maybe I'm not understanding how Unity works, but I don't think we should be calling it "the Unity".... just "Unity"?

("React")

File: src/content/developers/tutorials/Using-UseDApp-with-unity/index.md [hunk 15]
```diff
@@ -0,0 +1,206 @@
...
+Create a react app that use UseDApp,with UseDapp is possible easy connect you web wallet in a dapp to send transaction and get information about your account,
+also use [react-unity-webgl](https://github.com/jeffreylanters/react-unity-webgl) to help use the Unity in a react app.  
+
+### 1\.Create a react app and install the necessary plugins {#React-app-modules}
+Create a react app and install UseDApp and react-unity-webgl
+ 1. ```npx create-react-app dapp --template typescript ```
+ 2. ```npm install @usedapp/core```
+ 3. ```npm install react-unity-webgl@8.x```
+
+### 2\. Build unity project and add to react app{#add-unity-app}
```
Comment (wackerow) [inline src/content/developers/tutorials/Using-UseDApp-with-unity/index.md:101]: nix the slash, space before curly brackets

File: src/content/developers/tutorials/Using-UseDApp-with-unity/index.md [hunk 16]
```diff
@@ -0,0 +1,206 @@
...
+In your react app in the```public```folder create a new folder ```unity``` and build your unity project to this folder ```public/unity```.
```
Comment (wackerow) [inline src/content/developers/tutorials/Using-UseDApp-with-unity/index.md:102]: spaced, capitalization

File: src/content/developers/tutorials/Using-UseDApp-with-unity/index.md [hunk 17]
```diff
@@ -0,0 +1,206 @@
...
+![Captura de Tela (87)](https://user-images.githubusercontent.com/52639395/119470093-5ce34980-bd1e-11eb-8737-5d20aa36e6fa.png)
+
+### 3\. React unity{#react-unity-webgl}
+Now the unity project are builded ,use [react-unity-webgl](https://github.com/jeffreylanters/react-unity-webgl) to insert a unity project inside a react app, in ```src/App.tsx``` add this code.
```
Comment (wackerow) [inline src/content/developers/tutorials/Using-UseDApp-with-unity/index.md:106]: spacing, capitalization

File: src/content/developers/tutorials/Using-UseDApp-with-unity/index.md [hunk 18]
```diff
@@ -0,0 +1,206 @@
...
+```
+Set a size for unity WebGL using css in ```src/index.css```
+```css
+canvas{
+  width: 840px;
+  height: 480px;
+}
+
+```
+### 4\. Configure UseDApp{#UseDApp-config}
```
Comment (wackerow) [inline src/content/developers/tutorials/Using-UseDApp-with-unity/index.md:136]: same as above

File: src/content/developers/tutorials/Using-UseDApp-with-unity/index.md [hunk 19]
```diff
@@ -0,0 +1,206 @@
...
+Let's use the UseDApp to get the account balance and send a transction,but first we need to configure it in ```src/index.tsx```
```
Comment (wackerow) [inline src/content/developers/tutorials/Using-UseDApp-with-unity/index.md:137]: spacing

File: src/content/developers/tutorials/Using-UseDApp-with-unity/index.md [hunk 20]
```diff
@@ -0,0 +1,206 @@
...
+  <React.StrictMode>
+    <DAppProvider config={config}>
+      <App />
+    </DAppProvider>
+  </React.StrictMode>,
+  document.getElementById('root')
+);
+```
+### 5\. UseDApp to communicate with unity{#UseDApp}
+Create the functions that will interact with unity, in ```sr/App.tsx``` let's write  the first part of code to interact with unity.
```
Comment (wackerow) [inline src/content/developers/tutorials/Using-UseDApp-with-unity/index.md:159]: capitalization, typos ("sr/App.tsx"), double check spacing

File: src/content/developers/tutorials/Using-UseDApp-with-unity/index.md [hunk 21]
```diff
@@ -0,0 +1,206 @@
...
+    <div>
+      <Unity unityContext={unityContext} />
+    </div>
+  </div>
+    
+}
+```
+Run your app: ```npm run start```,
+
+When clicked in connect the UsaDApp will open your wallet and ask to connect to website,if you accept the connection, your account balance will be shown in the unity.
```
Comment (wackerow) [inline src/content/developers/tutorials/Using-UseDApp-with-unity/index.md:202]: Clicking "Connect" triggers UseDApp to open your wallet and ask to connect to the website. Accepting the connection allows your account balance to be shown within Unity.

File: src/content/developers/tutorials/Using-UseDApp-with-unity/index.md [hunk 22]
```diff
@@ -0,0 +1,206 @@
...
+When clicked in **Send 0.001ETH** , the unity will pass the address inside the input field and the amount to **SendTranscationFromUnity** in your react app,
```
Comment (wackerow) [inline src/content/developers/tutorials/Using-UseDApp-with-unity/index.md:203]: Clicking **Send 0.001 ETH** will pass the address and amount shown to **SendTranscationFromUnity** in your React app, which will prompt a pop-up to confirm the transaction.

**Review by wackerow — CHANGES_REQUESTED** [review]
@lucasespinosa28 Thanks for the PR... left comments to try to get this cleaned up. 

Big focus should be on readability and clear steps for the reader to follow. Try to avoid run-on sentences being separated by multiple commas, and double check your capitalization on proper names such as "React". 

I have never used this service myself, so if anyone else would like to comment on some of the details it would certainly be appreciated.

**Comment by wackerow** [comment]
Going to close this from inactivity, @lucasespinosa28  please feel free to reopen and resume or reach out with any questions.
